dre-service/src/main/java/io/cdap/re/RuleRequest.java

```diff
@@ -21,39 +21,64 @@
 /**
  * Class description here.
  */
-public final class RuleRequest {
+final class RuleRequest {
+
   private final String id;
   private final String description;
   private final String when;
   private final List<String> then;
 
-  public RuleRequest(String id, String description, String when, List<String> then) {
+  RuleRequest(String id, String description, String when, List<String> then) {
     this.id = id;
     this.description = description;
     this.when = when;
     this.then = then;
   }
 
-  public String getId() {
+  /**
+   * Validates the state of this {@link RuleRequest} excluding the id.
+   */
+  void validate() {
+    if (description == null || description.trim().isEmpty()) {
+      throw new IllegalArgumentException(
+        String.format("Rule '%s' requires the mandatory field 'description'", id)
+      );
+    }
+
+    if (when == null || when.trim().isEmpty()) {
+      throw new IllegalArgumentException(
+        String.format("Rule '%s' requires mandatory field 'when'", id)
+      );
+    }
+
+    if (getThen().trim().isEmpty()) {
+      throw new IllegalArgumentException(
+        String.format("Rule '%s' requires the mandatory field 'then'", id)
+      );
+    }
+  }
+
+  String getId() {
     return id;
   }
 
-  public String getDescription() {
+  String getDescription() {
     return description;
   }
 
-  public String getWhen() {
+  String getWhen() {
     return when;
   }
 
-  public String getThen() {
+  String getThen() {
```

Comment:
Since this object can be created through deserialization and not the constructor, this method can throw a null pointer exception if the user does not specify 'then'. It needs to check for null. 

I also don't know why this converts a List to a String. How is the string used? Can we change it to just return the List?

Reply:
The string is used to represent the list of actions of a `Rule`. Maybe it is changed to a string to pretty print the actions?

I can change it to a `List<String>`